clicommand/agent_start.go

```diff
@@ -65,6 +66,14 @@ type AgentStartConfig struct {
 	MetaDataGCP     bool     `cli:"meta-data-gcp" deprecated-and-renamed-to:"TagsFromGCP"`
 }
 
+func DefaultShell() string {
+	if runtime.GOOS == "windows" {
+		return `C:\Windows\System32\CMD.EXE /S /C`
+	} else {
+		return `/bin/bash -e -c`
+	}
+}
+
```

Comment:
Do you feel like adding another block here? or maybe it's time to use a switch?

```
if runtime.GOOS == "freebsd" { 
return `/usr/local/bin/bash -e -c`
} else if {
...
```

or
```
switch runtime.GOOS {
case "windows":
	shell_path = `C:\Windows\System32\CMD.exe /S /C`
case "freebsd", "openbsd", "netbsd":
    shell_path = `/usr/local/bin/bash -e -c`
default:
  # assume linux or is it time to add illumos/smartos/... ;-)
   shell_path = `/bin/bash -e -c`
}
...
```

BTW I'd expect there are a couple of other places, at present I "rewrite" this with a blunt object https://github.com/ideegeo/ports/blob/master/devel/buildkite-agent/Makefile#L30-L34

Reply:
Yeah that sounds pretty reasonable to me. Alternately we could just use `bash` and do a path lookup on it.

Reply:
Or maybe we just do drop back to `/bin/sh`.

Reply:
Nah, I think defaulting to /bin/sh would result in too many breakages for us.

Reply:
Added a switch.